cloudsync_onedrive.py

```diff
@@ -309,15 +193,14 @@ class OneDriveProvider(Provider):         # pylint: disable=too-many-public-meth
         scopes=['profile', 'openid', 'email', 'files.readwrite.all', 'sites.readwrite.all', 'offline_access'],
     )
 
-    _additional_invalid_characters = '#'
```

Comment:
Is the `#` not invalid anymore?

Reply:
this isn't used anywhere AFAICT - i went back in history ~2 years on this project, and also checked the current state of the base class in cloudsync

Reply:
This is used in the cloudsync project, in `cloudsync/tests/test_provider.py::test_special_characters`, so it is definitely important. If the tests are passing though, we should evaluate if, in fact, `#` is no longer invalid. Perhaps it was only invalid due to a bug in the SDK, and moving to the API allows us to support that character in filenames

Reply:
oh, i see -- it is passing though, and I confirmed that files/folders can have '#' in their names, so seems safe to remove.

Reply:
Sounds good